File: NN/VNNL/VNNL/VNNL/Activations/Activation_Sigmoid.cpp
#include "Activation_Sigmoid.h"

void Activation_Sigmoid::Forward(xt::xarray<double> inputs)
{
    this->inputs = inputs;
    this->outputs = 1 / (1 + xt::exp(-inputs));
}

void Activation_Sigmoid::Backward(xt::xarray<double> derivated_values)
{
    this->derivated_inputs = derivated_values * (1 - this->outputs) * this->outputs;
}
xt::xarray<double> Activation_Sigmoid::Prediction(xt::xarray<double> outputs)
{
    for (size_t index = 0 ; index <outputs.size();index++)
    {
        if(outputs[index] >0.5)
        {
            outputs[index] = 1;
        }
        else
        {
            outputs[index] = 0;
        }
    }
    return outputs;
}

File: NN/NNModule/Activations/ActivationSigmoid.py
import numpy as np
# Sigmoid activation

class Activation_Sigmoid:
    # Forward pass
    def forward(self, inputs):
        # Save input and calculate/save output
        # of the sigmoid function
        self.inputs = inputs
        self.output = 1 / (1 + np.exp(-inputs))
        # Backward pass

    def backward(self, derivated_values):
        # Derivative - calculates from output of the sigmoid function
        self.derivated_inputs = derivated_values * (1 - self.output) * self.output
    # Calculate predictions for outputs

    def predictions(self, outputs):
        return (outputs > 0.5) * 1
